### PR TITLE
Rename ics context

### LINKED ISSUE #245: Rename `Ics*` names to something more descriptive
We use this pattern a lot in the codebase, and I think we should move away from it. For example, `Ics20Context` could be renamed to `TransferContext`.

### PR DESCRIPTION
**Pull request by DaviRain-Su**
Closes: #245 

## Description

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests.
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/ibc/src/applications/transfer/context.rs [hunk 1]
```diff
@@ -93,16 +93,16 @@ pub trait BankKeeper {
 
 /// Captures all the dependencies which the ICS20 module requires to be able to dispatch and
 /// process IBC messages.
-pub trait Ics20Context:
-    Ics20Keeper<AccountId = <Self as Ics20Context>::AccountId>
-    + Ics20Reader<AccountId = <Self as Ics20Context>::AccountId>
+pub trait TransferContext:
```
Comment (plafer) [inline crates/ibc/src/applications/transfer/context.rs]: Thinking about this some more, I would prefer the names `TokenTransfer{Context,Keeper,Reader}`. @hu55a1n1 WDYT?

File: .changelog/unreleased/enhancement/245-rename-ics-context.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+- Rename Ics* names to something more descriptive
```
Comment (plafer) [inline .changelog/unreleased/enhancement/245-rename-ics-context.md]: Can we move this to `.changelog/unreleased/breaking-changes/` directory?

**Review by plafer — CHANGES_REQUESTED** [review]
Thank you again @DaviRain-Su!

Can we also rename the following:

+ `Ics18Context` -> `RelayerContext`
+ `Ics26Envelope` -> `MsgEnvelope`
    + And the variants (e.g. `Ics2Msg` -> `ClientMsg`, ...)

The only change left after this PR would be the directory names. However that's a really big breaking change (all imports will need to change); I think we should wait for #235 for that.

File: .changelog/unreleased/enhancement/245-rename-ics-context.md (hunk 2 above)
Comment (DaviRain-Su) [reply .changelog/unreleased/enhancement/245-rename-ics-context.md]: done

**Comment by DaviRain-Su** [comment]
> 

Done

File: crates/ibc/src/applications/transfer/context.rs (hunk 1 above)
Comment (DaviRain-Su) [reply crates/ibc/src/applications/transfer/context.rs]: done

**Review by plafer — APPROVED** [review]
Thank you!

**Pull request merged**